aexpect/client.py

```diff
@@ -1135,8 +1149,42 @@ def cmd_status_output(self, cmd, timeout=60, internal_timeout=None,
         else:
             raise ShellStatusError(cmd, o)
 
+    def cmd_status_output_safe(self, cmd, timeout=60):
+        """
+        Send a command and return its output and exit status (serial sessions).
+
+        In serial sessions, frequently the kernel might print debug or
+        error messages that make read_up_to_prompt to timeout. Let's try
+        to be a little more robust and send a carriage return, to see if
+        we can get to the prompt.
+
+        :param cmd: Command to send (must not contain newline characters)
+        :param timeout: The duration (in seconds) to wait for the prompt to
+                return
+
+        :return: A tuple (status, output) where status is the exit status and
+                output is the output of cmd
+        :raise ShellTimeoutError: Raised if timeout expires
+        :raise ShellProcessTerminatedError: Raised if the shell process
+                terminates while waiting for output
+        :raise ShellError: Raised if an unknown error occurs
+        """
+        o = self.cmd_output_safe(cmd, timeout)
+        try:
+            # Send the 'echo $?' (or equivalent) command to get the exit status
+            s = self.cmd_output_safe(self.status_test_command, 10)
```

Comment:
Well the problem of this is, that it won't work in case there is an output before/after the status:
```
0[3514161.846520] smpboot: Booting Node 0 Processor 7 APIC 0x7
[medic@w541 aexpect [1;31m[0m]$ 
```

anyway this should be quite rare and the only alternative I could see is something I wrote few years ago, but it's not portable to windows: https://github.com/autotest/autotest-docker/pull/192/commits/446ecf2302d20bb063e9653c7b2ebc6e905cd3c2#diff-2f87a33ba3bd086834255190bdff5b75R603

so I think we could go this way for now. I'd include a `:warning: This fails in case extra message is printed on the same line as the status output`, so people understand the limitation.